xapian-core/queryparser/queryparser.cc

```diff
@@ -172,6 +173,20 @@ QueryParser::parse_query(const string &query_string, unsigned flags,
     return result;
 }
 
+Query
+QueryParser::parse_math_query(const std::string & query_string, unsigned flags,
+			      const string &)
+{
+    if (query_string.empty()) return Query();
+
+    MathTermGenerator termgen;
+    auto query_terms = termgen.get_symbol_pair_list(query_string);
+    if (flags && FLAG_BOOLEAN)
```

Comment:
Do we have a boolean term in math ml?

FLAG_BOOLEAN is for distinguishing boolean and probabilistic term in normal term. 
Boolean terms always have wdf(word document frequency) of 1 and other normal terms can have higher wdf which is used in ranking.

Is there a plan to use that concept there and how we would describe boolean in math ml or math aware queries?

Reply:
Also, `flags && FLAG_BOOLEAN` is buggy as it's doing a boolean AND (`flags & FLAG_BOOLEAN` is a bit-wise AND).  As written, this just tests if flags is non-zero.

Reply:
> Boolean terms always have wdf(word document frequency) of 1 and other normal terms can have higher wdf which is used in ranking.

Minor correction here - boolean terms generally have wdf of 0 (there's no hard requirement as such but that's what `Xapian::Document::add_boolean_term()` does, and it makes sense that adding more filters shouldn't affect the document length, which Xapian defines as the sum of the wdf for all the terms indexing the document).

Reply:
Sorry, It is a mistake. It should have been bitwise `&` like olly pointed out and `FLAG_DEFAULT` instead of `FLAG_BOOLEAN`.  I pushed a fix by removing flag usage, as the default flag is sufficient for now.

As per usage of `FLAG_BOOLEAN`, I think `dice coefficient` weight metric performs set intersection, hence metric treats each math term as a `BOOLEAN` for the score computation right?

For the following case flag can be used I think:
Given query can be searched with or without unification of symbols. For this case, can we use flag to set the option?